code/addons/viewport/src/Tool.tsx

```diff
@@ -152,11 +142,13 @@ export const ViewportTool: FC = memo(
     }, [viewports]);
 
     useEffect(() => {
-      setState({
-        selected:
-          defaultViewport || (viewports[state.selected] ? state.selected : responsiveViewport.id),
-        isRotated: defaultOrientation === 'landscape',
-      });
+      if (state.selected) {
+        setState({
+          selected:
+            defaultViewport || (viewports[state.selected] ? state.selected : responsiveViewport.id),
+          isRotated: defaultOrientation === 'landscape',
+        });
+      }
```

Comment:
I think this might break the implementation. It would be safer to do something like:
```ts
state.selected && viewports[state.selected] ? state.selected : responsiveViewport.id
```

Reply:
My bad. I wrote the new implementation. Let me know what you think.